std/http/io.ts

```diff
@@ -258,7 +258,13 @@ export async function writeResponse(
   const headers = r.headers;
 
   for (const [key, value] of headers) {
-    out += `${key}: ${value}\r\n`;
+    if (key !== "set-cookie") {
+      out += `${key}: ${value}\r\n`;
+    } else {
+      value.split(/(?<!; expires\=\w+), /).forEach((v: string) => {
+        out += `${key}: ${v}\r\n`;
+      });
```

Comment:
I don't think this regex is the optimal way of achieving this...

Reply:
I am with you. No extra parsing should be necessary. But using the Headers construction with `string[][]` as parameter multiple headers get concatenated with `, `. You need some way to split it again, but not split within the expire value. (See comment in associated issue)

Reply:
I found the same issue yesterday and started working on it, didn't know there was already work being done. I have a working implementation with a different approach, I'll reference this PR.

Reply:
#4840